ambry-replication/src/main/java/com.github.ambry.replication/ReplicaThread.java

```diff
@@ -763,6 +768,9 @@ private void writeMessagesToLocalStoreAndAdvanceTokens(List<ExchangeMetadataResp
                   notification.onBlobReplicaCreated(dataNodeId.getHostname(), dataNodeId.getPort(),
                       messageInfo.getStoreKey().getID(), BlobReplicaSourceType.REPAIRED);
                 }
+                if (messageInfo.isTtlUpdated()) {
```

Comment:
Why not handle this case  in `processReplicaMetadataResponse`? 
The `else` block from line 587 to 609 looks like the right place.

Reply:
The blob is not yet present at that point. This TTL update is applied after the blob is put into the store.

Reply:
I see. It 'may' not yet present, right?

Reply:
Any value in doing it this way, rather than constructing a single "compacted" record and putting it?

Reply:
I thought about that but that would basically be have to be done by a `Transformer` like entity (for e.g. I could code this into the `BlobIdTransformer`) but I those impls are pluggable and removing that transformer or substituting will affect correctness in a way that is not obvious. So I decided to it this way since this is independent of any `Transformer` impls. If there is a reliable way of compacting this (without creating more copies in memory), I can consider that (in this or a subsequent PR)